java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/diff/ObfRegionSplitter.java

```diff
@@ -165,14 +177,143 @@ private Map<String, TLongObjectHashMap<RouteDataObject>> splitRegionRouteData(Ob
 							mp = new TLongObjectHashMap<>();
 							result.put(dw, mp);
 						}
+						if (heightData != null) {
+							attachElevationData(obj, heightData);
+							count++;
+						}
 						mp.put(obj.getId(), obj);
 					}
 				}
 			}
 		}
+		long t = (System.currentTimeMillis() - time) / 1000L;
+		long p = count / t;
+		System.out.println("Attach elevation data to ROUTE DATA section.");
+		System.out.println("Time total:" + t + " count:" + count + " per/sec:" + p);
 		return result;
 	}
 
+	private void attachElevationData(RouteDataObject obj, IndexHeightData heightData) {
+		List<Node> nodes = new ArrayList<>();
+		int id = 1;
+		for(int i = 0; i < obj.getPointsLength(); i++, id++) {
+			double lon = MapUtils.get31LongitudeX(obj.getPoint31XTile(i));
+			double lat = MapUtils.get31LatitudeY(obj.getPoint31YTile(i));
+			Node n = new Node(lat, lon, id);
+			nodes.add(n);
+		}
+
+		// Prepare Way with Nodes
+		Map<String, String> cachedNodeTags = new HashMap<>();
```

Comment:
This code is incorrect for multiple keys and unnecessary complicated. We know in advance which tags / keys could be added by elevation:
```
    public static final String ELE_ASC_START = "osmand_ele_start";
	public static final String ELE_ASC_END = "osmand_ele_end";
	public static final String ELE_INCLINE = "osmand_ele_incline_";
	public static final String ELE_INCLINE_MAX = "osmand_ele_incline_max";
	public static final String ELE_DECLINE = "osmand_ele_decline_";
	public static final String ELE_DECLINE_MAX = "osmand_ele_decline_max";
	public static final String ELE_ASC_TAG = "osmand_ele_asc";
	public static final String ELE_DESC_TAG = "osmand_ele_desc";
```